### PR TITLE
Select the current pull request

### LINKED ISSUE #2873: Select current pull request by default
The pull request list should have the pull request for the current branch selected by default.

### PR DESCRIPTION
**Pull request by joshaber**
Fixes #2873

### CODE REVIEW

**Comment by joshaber** [comment]
This is ready 🙏

File: app/src/ui/branches/pull-request-list.tsx [hunk 1]
```diff
@@ -27,6 +27,9 @@ interface IPullRequestListProps {
   /** The pull requests to display. */
   readonly pullRequests: ReadonlyArray<PullRequest>
 
+  /** The pull request associated with the current branch. */
```
Comment (niik) [inline app/src/ui/branches/pull-request-list.tsx]: Could we please add something like 'this is used to pre-select the currently checked out PR in the list of pull requests' to make it a little more descriptive?

File: app/src/ui/branches/pull-request-list.tsx [hunk 2]
```diff
@@ -48,19 +51,33 @@ export class PullRequestList extends React.Component<
...
+    ) {
       return
     }
 
-    this.setState({ groupedItems: [createListItems(nextProps.pullRequests)] })
+    const group = createListItems(nextProps.pullRequests)
+    const pullRequest = nextProps.currentPullRequest
+    const selectedItem = pullRequest
+      ? findItemForPullRequest(group, pullRequest)
+      : this.state.selectedItem
```
Comment (niik) [inline app/src/ui/branches/pull-request-list.tsx]: Can we really trust `this.state.selectedItem` to exist in the newly created list? It seems we'd need to do a lookup here comparing on `.id` as well, no?

**Comment by joshaber** [comment]
🇻🇦

File: app/src/ui/branches/pull-request-list.tsx [hunk 3]
```diff
@@ -48,19 +54,35 @@ export class PullRequestList extends React.Component<
...
+      nextProps.currentPullRequest === this.props.currentPullRequest
+    ) {
       return
     }
 
-    this.setState({ groupedItems: [createListItems(nextProps.pullRequests)] })
+    const group = createListItems(nextProps.pullRequests)
+    const pullRequest = nextProps.currentPullRequest
+    const selectedItem = pullRequest
+      ? findItemForPullRequest(group, pullRequest)
```
Comment (niik) [inline app/src/ui/branches/pull-request-list.tsx]: Sorry if I'm being picky here but won't this potentially steal focus when background updates happen? Thinking about the scenario where I'm using my keyboard in the PR list to move around and a background update happens which creates new pull request objects but the current pull request id hasn't changed. Should we perhaps introduce an equality function here which compares on `.id` instead?

**Comment by joshaber** [comment]
🐲 

As we discussed in chat, I _think_ we only wanna set the selected item to the `currentPullRequest` when the component is first created. After that, we don't wanna change it out from under the user in case they've changed selection manually.

**Pull request merged**